app/models/miq_worker/runner.rb

```diff
@@ -456,6 +484,17 @@ def set_process_title
 
   private
 
+  def worker_options
+    {
+      :messaging => MiqQueue.messaging_client_options,
+      :settings  => Settings.to_hash
+    }
+  end
+
+  def worker_cmdline
+    raise NotImplementedError, _("worker_cmdline must be implemented in a subclass")
+  end
+
```

Comment:
I can't comment on the code for `skip_heartbeat?` below but do we need to have some of these methods defined in the non rails worker runner classes instead of inheriting existing behavior?  I feel like they should opt-out of management explicitly but didn't see any mention on our traditional monitoring.

Reply:
@agrare my only remaining question is above, I'm good with going forward with this but was unsure how these non-rails workers explicitly disconnect themselves from legacy "worker" management.

Reply:
@jrafanie yes this is a good question, for the most part I think they are not disconnected from the standard worker management because they are still expected to heartbeat with sd_notify and the resource limits are enforced by systemd/k8s.

I think the only one which is different is settings reloading where the workers reload their `::Settings` hash in memory, this isn't feasible to do without connecting to the database so we will have to restart workers on settings changes.  We can do that in this PR or in a follow-up